datajunction-server/datajunction_server/api/graphql/main.py

```diff
@@ -110,6 +113,10 @@ class Query:
         resolver=log_resolver(common_dimensions),
         description="Get common dimensions for one or more nodes",
     )
+    downstream_nodes: list[Node] = strawberry.field(
```

Comment:
What do you think about potentially putting this as a `children` (or `downstreams`) attribute on the `Node` entity? I ask this because then we don't need a separate GraphQL query to get this info, and then you can always explore a `Node`'s downstreams.

Reply:
I don't know GraphQL well enough to have an opinion. But my thinking was that... this query (to find all downstream nodes) is expensive so I would prefer if people made it consciously, rather than add an attribute to `findNodes` and potentially affect the system with slow queries and maybe even have this data pulled in and not be used.

Reply:
Hmm, I think if we're worried about the query being expensive, putting it in a separate query doesn’t really protect us? I wonder if we can instead add some further protections like an argument to `downstreams` to control the depth fetched, so that even if someone does unintentionally fetch it on a `Node`, they will only fetch the most immediate layer of downstreams:
```
downstreams(depth: int = 1)
```

Reply:
Yeah, that makes sense. But let's add it later, if that's ok with you?

Reply:
Yeah, that works for me!